pandas/core/strings.py

```diff
@@ -74,11 +74,11 @@ def cat_core(list_of_columns: List, sep: str):
     """
     if sep == "":
         # no need to interleave sep if it is empty
-        arr_of_cols = np.asarray(list_of_columns, dtype=object)
+        arr_of_cols = np.asarray(list_of_columns, dtype=np.dtype(object))
```

Comment:
You don't need to revert this to be clear, but I also don't think it's needed to change this in *numpy* functions

Reply:
fair enough; i think in this case this was a find-and-replace